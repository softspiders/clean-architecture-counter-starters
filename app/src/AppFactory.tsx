import React, { createContext } from 'react'
import usePresenter from './adapters/presenters/usePresenter'
import App from './ui/App'
import { AdapterFactory } from './adapters'

interface ContextProps {
  counterPresenter: any
}

export const AppContext = createContext({} as ContextProps)

const adapterFactory = new AdapterFactory()

export const AppFactory = (): JSX.Element => {
  const counterPresenter = usePresenter(adapterFactory.getCounterUseCaseIn())
  return (
    <AppContext.Provider value={{ counterPresenter }}>
      <App />
    </AppContext.Provider>
  )
}
