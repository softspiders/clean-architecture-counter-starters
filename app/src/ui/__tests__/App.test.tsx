import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { AppFactory } from '../../AppFactory'

describe('AppFactory', () => {
  afterEach(cleanup)
  it('Should match the snapshot', () => {
    const { asFragment } = render(<AppFactory />)
    expect(asFragment()).toMatchSnapshot()
  })
})
