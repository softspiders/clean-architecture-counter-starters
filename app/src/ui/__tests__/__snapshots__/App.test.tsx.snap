// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFactory Should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="value-testid"
    >
       undefined 
      <button
        data-testid="button-testid"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
